lib/fastlane/plugin/ddg_apple_automation/actions/asana_find_release_task_action.rb

```diff
@@ -0,0 +1,189 @@
+require "fastlane/action"
+require "fastlane_core/configuration/config_item"
+require "httparty"
+require "json"
+require "octokit"
+require "time"
+require_relative "../helper/ddg_apple_automation_helper"
+require_relative "../helper/github_actions_helper"
+
+module Fastlane
+  module Actions
+    class AsanaFindReleaseTaskAction < Action
+      @constants = {}
+
+      def self.setup_constants(platform)
+        case platform
+        when "ios"
+          @constants = {
+            repo_name: "duckduckgo/ios",
+            release_task_prefix: "iOS App Release",
+            hotfix_task_prefix: "iOS App Hotfix Release",
+            release_section_id: "1138897754570756"
+          }
+        when "macos"
+          @constants = {
+            repo_name: "duckduckgo/macos-browser",
+            release_task_prefix: "macOS App Release",
+            hotfix_task_prefix: "macOS App Hotfix Release",
+            release_section_id: "1202202395298964"
+          }
+        end
+      end
+
+      def self.run(params)
+        asana_access_token = params[:asana_access_token]
+        github_token = params[:github_token]
+        platform = params[:platform] || Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+        setup_constants(platform)
+
+        latest_marketing_version = find_latest_marketing_version(github_token)
+        release_task_id = find_release_task(latest_marketing_version, asana_access_token)
+
+        release_task_url = Helper::DdgAppleAutomationHelper.asana_task_url(release_task_id)
+        release_branch = "release/#{latest_marketing_version}"
+        UI.success("Found #{latest_marketing_version} release task: #{release_task_url}")
+
+        Helper::GitHubActionsHelper.set_output("release_branch", release_branch)
+        Helper::GitHubActionsHelper.set_output("release_task_id", release_task_id)
+        Helper::GitHubActionsHelper.set_output("release_task_url", release_task_url)
+
+        {
+          release_task_id: release_task_id,
+          release_task_url: release_task_url,
+          release_branch: release_branch
+        }
+      end
+
+      def self.find_latest_marketing_version(github_token)
+        client = Octokit::Client.new(access_token: github_token)
+
+        # NOTE: `client.latest_release` returns release marked as "latest", i.e. a public release
+        latest_internal_release = client.releases(@constants[:repo_name], { per_page: 1 }).first
+
+        version = extract_version_from_tag_name(latest_internal_release&.tag_name)
+        if version.to_s.empty?
+          Fastlane::UI.user_error!("Failed to find latest marketing version")
+          return
```

Comment:
It will, but fastlane will break on the user_error, and rspec stubs this method when testing other methods, and for testing this specific method it's exactly what we need :)